src/main/scala/ai/privado/languageEngine/java/processor/JavaProcessor.scala

```diff
@@ -158,7 +159,15 @@ object JavaProcessor {
       println(s"${Calendar.getInstance().getTime} - Downloading dependencies and Parsing source code...")
     else
       println(s"${Calendar.getInstance().getTime} - Parsing source code...")
-    cpgconfig = Config(inputPath = sourceRepoLocation, fetchDependencies = !config.skipDownloadDependencies)
+
+
+
+    if (!Files.exists(Paths.get("./.privado"))) {
```

Comment:
@karan-batavia As this is a repeated function across, expose this as a function and use the function wherever required. Also test out this implementation in docker version